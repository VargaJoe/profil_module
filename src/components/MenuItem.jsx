import React, { Component } from 'react';

class MenuItem extends Component {

	constructor(props) {
	    super(props);
	    this.state = {};
  	}

	handleClick() {
		console.log(this.props.name + " is not active now!!!");
	}

	render() {

		return (
			<div className="sn_sidebar__menu__item" onClick={this.handleClick.bind(this)}>
				<span className="sn_sidebar__menu__item__icon">
					<i className={this.props.icon}></i>
				</span>
				<span className="sn_sidebar__menu__item__name">
					{this.props.name}
				</span>
			</div>
		);
	}
};

export default MenuItem;
