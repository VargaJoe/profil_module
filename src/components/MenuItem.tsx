import * as React				 			from 'react';
import { Link } 					        from 'react-router-dom';

interface Props {
	pathTo:  string;
	name: string;
	icon: string;
}

class MenuItem extends React.Component<Props, {}> {

	constructor(props: any) {
		super(props);

		this.handleClick = this.handleClick.bind(this);
	}

	handleClick () { 
		console.log( this.props.name);
	}

	render() {
		let path = this.props.pathTo ? '/' : this.props.pathTo;
		return (
			<Link to={path} className="sn_sidebar__menu__item" onClick={this.handleClick}>
				<span className="sn_sidebar__menu__item__icon">
					<i className={this.props.icon} />
				</span>
				<span className="sn_sidebar__menu__item__name">
					{this.props.name}
				</span>
			</Link>
		);
	}
}

export default MenuItem;
