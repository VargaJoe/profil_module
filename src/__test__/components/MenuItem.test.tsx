import * as React 						from 'react';
import MenuItem 				        from '../../components/MenuItem';
import { 
	configure, 
    shallow,
    mount } 							from 'enzyme';
import * as Adapter 					from 'enzyme-adapter-react-16';
import toJson 				      		from 'enzyme-to-json';

configure( {adapter: new Adapter()} );

describe('<MenuItem /> shallow rendering', () => {
    const menuItem = shallow(<MenuItem pathTo="/" name="home" icon="fi "/>);
    // test Snapshot 
	it('Match to snapshot', () => {
		expect(toJson(menuItem)).toMatchSnapshot();
    });
});
 