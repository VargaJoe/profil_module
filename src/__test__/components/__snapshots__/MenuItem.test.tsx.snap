// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> shallow rendering Match to snapshot 1`] = `
<Link
  className="sn_sidebar__menu__item"
  onClick={[Function]}
  replace={false}
  to="/"
>
  <span
    className="sn_sidebar__menu__item__icon"
  >
    <i
      className="fi "
    />
  </span>
  <span
    className="sn_sidebar__menu__item__name"
  >
    home
  </span>
</Link>
`;

exports[`<MenuItem /> shallow rendering Match to snapshot 2`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <MenuItem
      icon="fi "
      name="home"
      pathTo="/"
    >
      <Link
        className="sn_sidebar__menu__item"
        onClick={[Function]}
        replace={false}
        to="/"
      >
        <a
          className="sn_sidebar__menu__item"
          href="/"
          onClick={[Function]}
        >
          <span
            className="sn_sidebar__menu__item__icon"
          >
            <i
              className="fi "
            />
          </span>
          <span
            className="sn_sidebar__menu__item__name"
          >
            home
          </span>
        </a>
      </Link>
    </MenuItem>
  </Router>
</BrowserRouter>
`;
